packages/platform-server/src/provide_server.ts

```diff
@@ -0,0 +1,38 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {HttpClientModule} from '@angular/common/http';
+import {EnvironmentProviders, importProvidersFrom, makeEnvironmentProviders} from '@angular/core';
+import {BrowserModule} from '@angular/platform-browser';
+import {NoopAnimationsModule} from '@angular/platform-browser/animations';
+
+import {PLATFORM_SERVER_PROVIDERS} from './server';
+
+/**
+ * Sets up providers necessary to enable server rendering functionality for the application.
+ *
+ * @usageNotes
+ *
+ * Basic example of how you can add server support to your application:
+ * ```ts
+ * bootstrapApplication(AppComponent, {
+ *   providers: [provideServerSupport()]
+ * });
+ * ```
+ *
+ * @publicApi
+ * @returns A set of providers to setup the server.
+ */
+export function provideServerSupport(): EnvironmentProviders {
+  return makeEnvironmentProviders([
+    importProvidersFrom(BrowserModule),
+    importProvidersFrom(HttpClientModule),
+    importProvidersFrom(NoopAnimationsModule),
```

Comment:
We can use standalone APIs here to add providers, WDYT? 
```suggestion
    // importProvidersFrom(BrowserModule), -- do we need it?
    provideHttpClient(withInterceptorsFromDi()),
    provideNoopAnimations(),
```

Reply:
- The `importProvidersFrom(BrowserModule.withServerTransition({appId}))` should not be set in platform-server, it needs to be set on the browser application. This is infact currently incorrect and we should remove it. 

- `withServerTransition` can be deprecated following https://github.com/angular/angular/pull/48253

- `importProvidersFrom(BrowserModule)` or `BROWSER_MODULE_PROVIDERS `, is needed, unless users will provide `importProvidersFrom(BrowserModule)` in their code, as currently `importProvidersFrom(BrowserModule)` is done internally in https://github.com/angular/angular/blob/5e7fc259ead62ee9b4f8a9a77a455065b6a8e2d8/packages/platform-browser/src/browser.ts#L118 where that platform will never be created when using `renderApplication` since this creates another platform.

Reply:
> The importProvidersFrom(BrowserModule.withServerTransition({appId})) should not be set in platform-server, it needs to be set on the browser application. This is infact currently incorrect and we should remove it. As the APP_ID needs to be set in browser application. Setting it in the renderApplication will cause the APP_ID to mismatch.
>
>withServerTransition can be deprecated following https://github.com/angular/angular/pull/48253 and instead the APP_ID can be provided as a token when needed. By default it can be the comp id of the root component or just ng.

That makes sense 👍 Let's do in a followup PR (plz let me know if you want me to help)!

> importProvidersFrom(BrowserModule) or BROWSER_MODULE_PROVIDERS , is needed, unless users will provide importProvidersFrom(BrowserModule) in their code, as currently importProvidersFrom(BrowserModule) is done internally in

The `BROWSER_MODULE_PROVIDERS` are added to the app providers (not platform providers), so if an app is bootstrapped using `bootstrapApplication`, those providers should be available. For SSR case the code would roughly be like this:

```
renderApplication(() => {
  // The `BROWSER_MODULE_PROVIDERS` are added here
  // (inside the `bootstrapApplication` call).
  return bootstrapApplication(RootCmp, {
    providers: [
      provideServerSupport(),
      // ... the rest of the providers
    ]
  });
});
```

I might be missing something though, please let me know what you think.

Reply:
Thanks @AndrewKushnir, I missed the part that `BROWSER_MODULE_PROVIDERS` are app and not platform providers.